lms/djangoapps/lti_provider/users.py

```diff
@@ -51,13 +60,32 @@ def create_lti_user(lti_user_id, lti_consumer):
     created = False
     while not created:
         try:
-            edx_user_id = generate_random_edx_username()
-            edx_email = "{}@{}".format(edx_user_id, settings.LTI_USER_EMAIL_DOMAIN)
+            if lti_email is not None:
+                edx_email=cut_to_max_len(lti_email, EMAIL_SIZE)
+                edx_username=cut_to_max_len(lti_email, USERNAME_SIZE)
+                try:
+                    #if this username already exists then use the random username
+                    edx_user = User.objects.get(username=edx_username)
```

Comment:
in users.py:LtiBackend.authenticate they search for an existing user in this manner so I thought there is nothing better